firebase-firestore/src/roboUtil/java/com/google/firebase/firestore/ValueHelper.java

```diff
@@ -0,0 +1,98 @@
+import com.google.firebase.firestore.model.DatabaseId;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firestore.v1.ArrayValue;
+import com.google.firestore.v1.Value;
+import com.google.protobuf.NullValue;
+import com.google.type.LatLng;
+import java.util.List;
+
+/** Test helper to create Firestore Value protos from Java types. */
+public class ValueHelper {
+
+  public static Value valueOf(Object o) {
+    if (o instanceof Value) {
+      return (Value) o;
+    } else if (o instanceof String) {
+      return (Value.newBuilder().setStringValue((String) o).build());
+    } else if (o instanceof Integer) {
+      return (Value.newBuilder().setIntegerValue((long) (Integer) o).build());
+    } else if (o instanceof Long) {
+      return (Value.newBuilder().setIntegerValue((Long) o).build());
+    } else if (o instanceof Double) {
+      return (Value.newBuilder().setDoubleValue((Double) o).build());
+    } else if (o instanceof Boolean) {
+      return (Value.newBuilder().setBooleanValue((Boolean) o).build());
+    } else if (o instanceof Timestamp) {
+      Timestamp timestamp = (Timestamp) o;
+      return (Value.newBuilder()
+          .setTimestampValue(
+              com.google.protobuf.Timestamp.newBuilder()
+                  .setSeconds(timestamp.getSeconds())
+                  .setNanos(timestamp.getNanoseconds())
+                  .build())
+          .build());
+    } else if (o instanceof GeoPoint) {
+      GeoPoint geoPoint = (GeoPoint) o;
+      return (Value.newBuilder()
+          .setGeoPointValue(
+              LatLng.newBuilder()
+                  .setLatitude(geoPoint.getLatitude())
+                  .setLongitude(geoPoint.getLongitude())
+                  .build())
+          .build());
+    } else if (o instanceof Blob) {
+      return (Value.newBuilder().setBytesValue(((Blob) o).toByteString()).build());
+    } else if (o instanceof DocumentReference) {
+      return (Value.newBuilder()
+          .setReferenceValue(
+              "projects/projectId/databases/(default)/documents/"
+                  + ((DocumentReference) o).getPath())
+          .build());
+    } else if (o instanceof List) {
+      ArrayValue.Builder list = ArrayValue.newBuilder();
+      for (Object element : (List) o) {
+        list.addValues(valueOf(element));
+      }
+      return (Value.newBuilder().setArrayValue(list).build());
+    } else if (o == null) {
+      return (Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build());
+    }
+
+    throw new UnsupportedOperationException();
+  }
+
+  public static Value map(Object... entries) {
+    com.google.firestore.v1.MapValue.Builder builder =
+        com.google.firestore.v1.MapValue.newBuilder();
+    for (int i = 0; i < entries.length; i += 2) {
+      builder.putFields((String) entries[i], valueOf(entries[i + 1]));
+    }
+    return Value.newBuilder().setMapValue(builder).build();
+  }
+
+  public static Value wrapRef(DatabaseId dbId, DocumentKey key) {
```

Comment:
The rest of this class doesn't really have much to do with wrapping, but does create values. How about `refValue` or something like that?

Reply:
Done